lib/game/game.dart

```diff
@@ -91,6 +91,12 @@ abstract class Game {
 
   VoidCallback pauseEngineFn;
   VoidCallback resumeEngineFn;
+
+  /// Use this method to load the assets need for the game instance to run
+  Future<void> onLoad() async {}
+
+  /// Returns the widget which will be show while the instance is loading
+  Widget loadingWidget() => Container();
```

Comment:
is this easily compatible with our splash screen? I really want to have a way to pass a Future for all my loads to splash screen. see my code on gravitational_waves for an example of how useful this would be!

Reply:
I think this don't replace a splash screen, and for the most of the cases the default empty Container will suffice. A splash screen would be more of loading things for the entire game, like initializing IAP, Ads, services and etc. And this the assets needed for that Game instance.

Reply:
I think we can load both on the splash screen. I mean, in general. of course we should be as flexible as possible.

Reply:
Exactly, we need to keep this flexible. On Snake Chef for example, I would have things being loaded on the splash, and things here on the game.

Reply:
yes, of course it is flexible, it can be any widget. I am just asking if it's easy to use the splash screen here if the user wants. they can provide sort of loading bar easily as well, of course.

Reply:
Oh, alright, I think now I understood your suggestion, yep, would be cool if the splash screen could be integrated somehow, we could probably think on something when we update the splash screen module to be compatible with the v1